gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableFileSet.java

```diff
@@ -21,20 +21,26 @@
 import java.util.Collection;
 import org.apache.gobblin.data.management.copy.CopyConfiguration;
 import org.apache.gobblin.data.management.copy.CopyEntity;
+import org.apache.gobblin.data.management.partition.FileSet;
+
 
 /**
- * A {@link IcebergFileSet} that generates {@link CopyEntity}s for a Hive Catalog based Iceberg table
+ * A {@link IcebergTableFileSet} that generates {@link CopyEntity}s for a Hive Catalog based Iceberg table
+ * A {@link FileSet} for Iceberg datasets containing information associated with an Iceberg table.
  */
-public class IcebergTableFileSet extends IcebergFileSet{
+public class IcebergTableFileSet extends FileSet<CopyEntity> {
 
   private final CopyConfiguration copyConfiguration;
+  private final IcebergDataset icebergDataset;
+
   public IcebergTableFileSet(String name, IcebergDataset icebergDataset, CopyConfiguration configuration) {
     super(name, icebergDataset);
     this.copyConfiguration = configuration;
+    this.icebergDataset = icebergDataset;
   }
 
   @Override
   protected Collection<CopyEntity> generateCopyEntities() throws IOException {
-    return this.getIcebergDataset().generateCopyEntitiesForTableFileSet(this.copyConfiguration);
+    return this.icebergDataset.generateCopyEntitiesForTableFileSet(this.copyConfiguration);
```

Comment:
more I see this, I'd prefer to shorten the name to `IcebergDataset.generateCopyEntities`.  whaddya say?